nerium/query.py

```diff
@@ -49,23 +45,30 @@ def extract_metadata(query_string):
     return metadata
 
 
-def parse_query_file(query_name):
-    """Parse query file and add query statement and metadata."""
-    query_obj = init_query(query_name)
-    try:
-        with open(query_obj.path) as f:
+def read_query_file(path):
+    if path.startswith("s3://"):
+        fs = s3fs.S3FileSystem(anon=False)
+        with fs.open(path, "rt") as f:
```

Comment:
The "rt" may fall back to "rb" but it's not actually supported in this library (or indeed the underlying S3 API), so I think this line is kind of misleading.  See a similar comment on a previous pr [here](https://github.com/nerium-data/sqla-raw/pull/17/files#r995248650).

Reply:
The documentation would seem to be out of date on this point

```
>>> f = s3fs.S3FileSystem()
>>> type(f.open('s3://nerium-test/test.sql', 'rb').read())
<class 'bytes'>
>>> type(f.open('s3://nerium-test/test.sql', 'rt').read())
<class 'str'>
```

But then again, I'm not crazy about relying on an undocumented feature either. I used `rt` here because I was getting TypeErrors from treating the read result as a string later, but so an explicit `str()` call on a binary read is probably better. I'll push that change before this gets merged.

Reply:
Perhaps this is a case of `s3fs` maintainers not understanding the `fsspec` library they're building on top of? It's true that S3 access is always binary, but `fsspec.OpenFile` [allows layering text-mode over it](https://filesystem-spec.readthedocs.io/en/latest/features.html#openfile-instances), using `PickleableTextIOWrapper`. None of this changes that an explicit conversion to text would be better here -- I just wanted to understand how this was working at all, so I dug around a bit

Reply:
Ah, thanks!  That makes more sense.  I had looked at the docs and the source for both `s3fs` and `S3File` but I skipped the `fsspec` docs and missed the bit about "Transparent text-mode".  It just looked a bit odd since I knew S3 access to be binary.

Reply:
Thank YOU! This was all pretty interesting.

Even though it works as originally written, I have gone ahead and changed this to an expressly binary read that I explicitly convert to text, for the sake of clarity for now. I'm also considering opening an issue on `s3fs` to clarify their documentation.